project-service/api.md

```diff
@@ -0,0 +1,594 @@
+  tasks: [
+    {
+      id: 1,
+      name: 'create api documentation',
+      status: 'In Progress',
+      userId: 0,
+      sprintId: 1
+    }
+  ]
+}
+```
+
+- Error Response
+  + Invalid Project
+    * Code: `404 NOT FOUND`
+    * Content: `{ error: "Project doesn't exist." }`
+  + Not Authenticated
+    * Code: `401 UNAUTHORIZED`
+    * Content: `{ error: "You are unauthorized to make this request."}`
+
+### Modify Project
+
+Edit an existing project.
+
+- URL
+  + `/projects`
+- Method
+  + `PUT`
+- URL Params
+  + Required
+    * `projectId=[number]`
+- Data Params
+  + Optional
+    * `name=[string]`
+- Success Response
+  + Code: `200 OK`
+  + Content:
+
+```
+{
+  id: 1,
+  name: 'turtle',
+  createdAt: [date],
+  updatedAt: [date]
+}
+```
+
+- Error Response
+  + Invalid Project
+    * Code: `404 NOT FOUND`
+    * Content: `{ error: "Project doesn't exist." }`
+  + Not Authenticated
+    * Code: `401 UNAUTHORIZED`
+    * Content: `{ error: "You are unauthorized to make this request."}`
+
+### Add User to Project
+
+Add a user to a project
+
+- URL
+  + `/projects/users`
+- Method
+  + `POST`
+- URL Params
+  + None
+- Data Params
+  + Required
+    * `projectId=[number]`
+    * `userId=[number]`
+- Success Response
+  + Code: `204 NO CONTENT`
+- Error Response
+  + Invalid Project
+    * Code: `404 NOT FOUND`
+    * Content: `{ error: "Project doesn't exist." }`
+  + Invalid User
+    * Code: `404 NOT FOUND`
+    * Content: `{ error: "User doesn't exist." }`
+  + Not Authenticated
+    * Code: `401 UNAUTHORIZED`
+    * Content: `{ error: "You are unauthorized to make this request."}`
+
+### Remove User from Project
+
+Remove a user to a project
+
+- URL
+  + `/projects/users`
+- Method
+  + `DELETE`
+- URL Params
+  + None
+- Data Params
+  + Required
+    * `projectId=[number]`
+    * `userId=[number]`
+- Success Response
+  + Code: `204 NO CONTENT`
+- Error Response
+  + Invalid Project
+    * Code: `404 NOT FOUND`
+    * Content: `{ error: "Project doesn't exist." }`
+  + Invalid User
+    * Code: `404 NOT FOUND`
+    * Content: `{ error: "User doesn't exist." }`
+  + Not Authenticated
+    * Code: `401 UNAUTHORIZED`
+    * Content: `{ error: "You are unauthorized to make this request."}`
+
+## Sprints
+
+### Fetch Sprints
+
+Get all existing sprints for a project
+
+- URL
+  + `/sprints`
+- Method
+  + `GET`
+- URL Params
+  + Required
+    * `projectId=[number]`
+- Data Params
+  + None
+- Success Response
+  + Code: `200 OK`
+  + Content:
+
+```
+[
+  {
+    id: 1,
+    name: 'mvp'
+    status: 'Started',
+    startDate: [date],
+    endDate: [date]
+  }, {
+    id: 2,
+    name: 'profit',
+    status: 'Not Started',
+    startDate: [date],
+    endDate: [date]
+  }
+]
+```
+
+- Error Response
+  + Invalid Project
+    * Code: `404 NOT FOUND`
+    * Content: `{ error: "Project doesn't exist." }`
+  + Not Authenticated
+    * Code: `401 UNAUTHORIZED`
+    * Content: `{ error: "You are unauthorized to make this request."}`
+
```

Comment:
Add fetching a sprint (with its associated tasks and, I guess, rudimentary information about users that are assigned to tasks)